### PR TITLE
add support for specifying container registry

### PR DESCRIPTION
**Pull request by philip-mannes-cardano**
Please help, we're facing rate throttling by Dockerhub due to this action. Please allow us to specify the registry.
- addresses https://community.sonarsource.com/t/sonarcloud-github-action-registry-hitting-rate-limit/99321/19, please adjust as needed

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,5 @@
-FROM sonarsource/sonar-scanner-cli:11.1
+ARG CONTAINER_REGISTRY
+FROM ${CONTAINER_REGISTRY}/sonarsource/sonar-scanner-cli:11.1
```
Comment (marciorgb) [inline Dockerfile:2]: Maybe letting the leading `/` in the `CONTAINER_REGISTRY` arg and adding a default value as ""

Comment (philip-mannes-cardano) [reply Dockerfile]: New scans seem to not pass using this rewrite, could someone please advise? Dockerhub throttling caused by this action is blocking all our CICD processes.

image

**Comment by henryju** [comment]
Hi,

Thanks for your contribution. We are aware of the rate limit issue and are willing to replace this action with a composite action without relying on Docker.
See for example https://github.com/SonarSource/sonarqube-scan-action/tree/jh/experiment_composite_action

This is likely to come before the end of the year.

**Comment by antonioaversa** [comment]
Hi @philip-mannes-cardano, 
We have just released `sonarqube-scan-action@v4.1.0`. `v4.0.0` removes Docker, moving to a composite action, while `v4.1.0` brings several improvements to the action, which is now the single entrypoint for SonarQube Server and Cloud (formerly SonarCloud).

You can read about these new releases 
- `v4.0.0` in [our community post](https://community.sonarsource.com/t/the-upcoming-release-of-github-action-for-sonarqube-v-4-0-0-remove-docker/130180) or in the [GitHub release notes](https://github.com/SonarSource/sonarqube-scan-action/releases/tag/v4.0.0)
- `v4.1.0`: in [our community post](https://community.sonarsource.com/t/the-new-release-of-github-action-for-sonarqube-v-4-1-0-server-and-cloud/131400) or in the [GitHub release notes](https://github.com/SonarSource/sonarqube-scan-action/releases/tag/v4.1.0)

`sonarqube-scan-action@v4.1.0` should be a drop-in replacement. Please, give it a try and let us know.

**Comment by pierre-guillot-gh** [comment]
Closing the PR as we won't merge this contribution, as we migrated to composite action. Feel free to continue the conversation here.

**Pull request closed without merging**